modules/pam_mkhomedir/mkhomedir_helper.c

```diff
@@ -142,6 +144,15 @@ create_homedir(const struct passwd *pwd,
       if (S_ISDIR(st.st_mode))
       {
          retval = create_homedir(pwd, newsource, newdest);
+         if (dir_mode) {
+            if (retval == PAM_SUCCESS &&
+            chmod(newdest, st.st_mode & (~u_mask)) != 0)
```

Comment:
This is not the right place to invoke `chmod` because `create_homedir` already invokes `chmod` for this directory.